docs/coding-style.md

```diff
@@ -218,6 +218,10 @@ It can still be used when and where it makes sense. For instance, when a class h
 implementing generic interfaces (such as `IComparable`, `IDisposable`), it can make sense to have regions 
 for the implementation of these interfaces.
 
+## ValueTuples
+
+Do not use `ValueTuples` in production code. The usage in test projects is fine. `ValueTuples` are not supported in MsBuild 14 and while MsBuild 14 is not officially supported anymore, we still don't want to break it, if we can avoid it.
```

Comment:
We need to verify if this applies to all tuples, or only named tuples. I have a weak memory that it's only the second, but I'm not sure

Reply:
Would this change anything for our decision?

Reply:
Educational: What caused `System.ValueTuple` to be available now? We couldn't use that type before. Did we remove some older target framework? If so, we are walking on thin ice if we try to support frameworks without compiling against them.

Reply:
To help answer @pavel-mikula-sonarsource's question: unlike `ValueTuple`, that is a struct introduced around .NET Framework 4.6, to support deconstruction, the `Tuple` classes,  were around since .NET Framework 4.0, that is MSBuild 4.7. 
So, I think we should be good with `Tuple`. Remark: I haven't checked all its many derivations, based on the number of elements in the tuple. I have only checked a couple...

Reply:
The problem is that `System.ValueType` is only part of .Net framework 4.7 upwards. For lower frameworks, it needs to be added as a nugget package.
Tuple syntax is problematic as it sometimes involves ValueTuple and sometimes not. If we want to properly support net462 as we claim, we need to compile against that framework. An analyzer or a style guide is not enough.
I also like to point out, that there might be other problems with net462 we are not aware of. We should at least compile against the framework in CI to catch any missing types/overloads on that framework.